dataset_loaders/images/polyps912.py

```diff
@@ -87,12 +90,46 @@ def __init__(self, which_set='train', *args, **kwargs):
         else:
             raise ValueError("Unknown set requested: %s" % which_set)
 
+        self.preload = preload
+
         # Define the images and mask paths
         self.image_path = os.path.join(self.path, self.which_set, 'images')
         self.mask_path = os.path.join(self.path, self.which_set, 'masks2')
 
+        if self.preload:
+            self._preload_data()
+
         super(Polyps912Dataset, self).__init__(*args, **kwargs)
 
+    def _preload_data(self):
+        """Preload all data in memory.
+
+        The images will be stored as a list of ndarrays in self.image,
+        the masks will be in self.mask, in the same order as
+        self.filename.
+        In addition, self.image_name_to_idx will contain a dictionary
+        mapping the root of the image name to its index.
+        """
+        from skimage import io
+        image_all = []
+        mask_all = []
+        image_name_to_idx = {}
+        for idx, img_name in enumerate(self.filenames):
+            image_name_to_idx[img_name] = idx
+            img = io.imread(os.path.join(self.image_path, img_name + ".bmp"))
```

Comment:
Can you move the code that loads images and masks (L119-L127) in a separate function `_load_image(image_batch, mask_batch, filename_batch, img_name, prefix=None)` and call that function from here and from `load_sequence`? This way the common code doesn't get duplicated. 

As a more general step, I think it might be worthwhile to do the same for all the datasets and have a `preload` flag in parallel_loader.py, but I don't know if you feel like going that far with this PR or not.